mem/buffers_test.go

```diff
@@ -0,0 +1,44 @@
+package mem
```

Comment:
Also, if possible, could we move these to package `mem_test` to ensure that we testing the API and not relying on anything internal to the package. Thanks.